File: jcrm-ui/src/app/entity/phone-number.ts
import {Country} from '@app/entity/country';
import {Entity} from '@app/entity/entity';

export class PhoneNumber extends Entity {
    constructor(
        public country: Country | null,
        public regionCode: string | null,
        public phoneNumber: string | null
    ) {
        super();
    }
}

File: ui_flutter/lib/src/jcrm/ui/flutter/entity/phone_number.dart
import 'package:jcrm_ui_flutter/src/jcrm/ui/flutter/entity/entity.dart';
import 'package:jcrm_ui_flutter/src/jcrm/ui/flutter/resource/phone_number_resource.dart';

class PhoneNumber extends Entity {

  final String countryCode;
  final String regionCode;
  final String phoneNumber;

  final int type;

  PhoneNumber(this.countryCode, this.regionCode, this.phoneNumber, this.type);

  PhoneNumber.fromResource(PhoneNumberResource resource)
      : countryCode = resource.countryCode,
        regionCode = resource.regionCode,
        phoneNumber = resource.phoneNumber,
        type = resource.type;
}